### ISSUE TITLE
Error in .new_IRanges_from_start_end(start, end) :  'start' or 'end' cannot contain NAs

### ISSUE THREAD

**Issue by albert-ying** [neurogenomics/MungeSumstats#83]
Maybe it is better to just remove those problematic SNPs (or infer from CHR and POS columns), instead of throwing errors?
```
[ins] r$> munged = MungeSumstats::format_sumstats(                                          
                path = file_processed,
                INFO_filter = 0.8,
                ref_genome = "GRCh37"
              )
|--------------------------------------------------|                                        
|==================================================|                                        
First line of summary statistics file: 
SNP     chr     pos     effect_allele   other_allele    EAF     b       se      p       N   
295 SNP IDs are not correctly formatted. These will be corrected from the reference genome. 
603305 SNP IDs appear to be made up of chr:bp, these will be replaced by their SNP ID from the reference genome
Error in .new_IRanges_from_start_end(start, end) : 
  'start' or 'end' cannot contain NAs
In addition: Warning message:
In eval(jsub, SDenv, parent.frame()) : NAs introduced by coercion
```

Some of SNP id looks like this in my file
<img width="118" alt="image" src="https://user-images.githubusercontent.com/59846322/150716479-f938bf5e-dd6c-4958-8e32-403569295dfe.png">

**Comment by Al-Murphy** [comment]
Hey Albert, I agree an error is not the way to go and generally an error won't be thrown. It appears the issue occurs here when mungsumstats tries to infer the rsid for a specific SNP who's current value is made up of chr:bp but when looking up that value from the reference genome something in the search breaks it. I think this is something to do with the POS and/or CHR having NA's but testing this I can't get it to fail (below runs fine):

```
library(MungeSumstats)
library(data.table)

eduAttainOkbayPth <- system.file("extdata", "eduAttainOkbay.txt",
                                 package = "MungeSumstats"
)

a <- fread(eduAttainOkbayPth)
a[MarkerName %in% c("rs12987662","rs9320913"),POS:=NA]
a[MarkerName %in% c("rs12987662","rs9320913"),CHR:=NA]
a[MarkerName=="rs12987662",MarkerName:="2:100821548_AC_A"]
a[MarkerName=="rs9320913",MarkerName:="6:98584733_AC_A"]
pth <- tempfile()
fwrite(a,pth,sep = "\t")

reformatted <- format_sumstats(
  path = pth,
  ref_genome = "GRCh37"
)

```

Can you share a small subset of a sumstats that this error appears on?

**Comment by albert-ying** [comment]
Hi Alan,

I can't locate the 

You can do `wget https://broad-ukb-sumstats-us-east-1.s3.amazonaws.com/round2/annotations/variants.tsv.bgz -O variants.tsv.bgz`

And make a pseudo-GWAS data from there.

```
variants = read_tsv("variants.tsv.bgz", col_types = "ccdccc---d--d------------")
pseudo_gwas = variants |>
  select(chr, pos, effect_allele = alt, other_allele = ref, EAF = AF, note = variant) |>
  mutate(b = 1, se = 1, p = 0.9, N = 500000) 
write_csv(pseudo_gwas, "./pseudo_gwas.csv")

munged = MungeSumstats::format_sumstats(
  path = "./pseudo_gwas.csv",
  INFO_filter = 0.8,
  ref_genome = "GRCh37"
)
```

You should be able to see the error I mentioned.

**Comment by albert-ying** [comment]
I was able to avoid the error by removing the SNP column. And the program is able to retrieve the SNP id from `CHR` and `POS`. Maybe it is better to get the rsid from `CHR` `POS` `A1` `A0` columns (since they are required anyway), instead of trying to parse the original SNP id?

**Comment by Al-Murphy** [comment]
Hey Albert,

Are you sure this is the correct dataset? I'm still not getting an error when running (MungeSumstats version 1.2.0):

```
> munged = MungeSumstats::format_sumstats(
+   path = "~/Downloads/pseudo_gwas.csv",
+   INFO_filter = 0.8,
+   ref_genome = "GRCh37"
+ )


******::NOTE::******
- Formatted results will be saved to `tempdir()` by default.
- This means all formatted summary stats will be deletedupon ending the R session.
- To keep formatted summary stats, change `save_path` ( e.g. `save_path=file.path('./formatted',basename(path))` ),  or make sure to copy files elsewhere after processing ( e.g. `file.copy(save_path, './formatted/' )`.
********************

Formatted summary statistics will be saved to ==> /var/folders/hd/jm8lzp7s4dl_wlkykzhz66x80000gn/T//Rtmp916qbx/file388825789f34.tsv.gz
Reading header.
Tabular format detected.
Importing tabular file: ~/Downloads/pseudo_gwas.csv
|--------------------------------------------------|
|==================================================|
Standardising column headers.
First line of summary statistics file: 
SNP	chr	pos	effect_allele	other_allele	EAF	note	b	se	p	N	
Summary statistics report:
   - 13,791,467 rows
   - 13,778,263 unique variants
   - 0 genome-wide significant variants (P<5e-8)
   - 23 chromosomes
Checking for multi-GWAS.
Checking for multiple RSIDs on one row.
Checking SNP RSIDs.
307 SNP IDs are not correctly formatted. These will be corrected from the reference genome.
603,614 SNP IDs appear to be made up of chr:bp, these will be replaced by their SNP ID from the reference genome
Checking for merged allele column.
Ensuring all SNPs are on the reference genome.
Loading reference genome data.
723,216 SNPs are not on the reference genome. These will be corrected from the reference genome.
Checking for correct direction of A1 (reference) and A2 (alternative allele).
There are 253 SNPs where neither A1 nor A2 match the reference genome. These will be removed.
There are 40,965 SNPs where A1 doesn't match the reference genome.
These will be flipped with their effect columns.
Reordering so first three column headers are SNP, CHR and BP in this order.
Reordering so the fourth and fifth columns are A1 and A2.
Checking for missing data.
Checking for duplicate columns.
Checking for duplicate SNPs from SNP ID.
30,997 RSIDs are duplicated in the sumstats file. These duplicates will be removed
Checking for SNPs with duplicated base-pair positions.
1 base-pair positions are duplicated in the sumstats file. These duplicates will be removed.
Filtering SNPs, ensuring SE>0.
Ensuring all SNPs have N<5 std dev above mean.
Removing 'chr' prefix from CHR.
Making X/Y CHR uppercase.
387,119 SNPs are on chromosomes X, Y, MT and will be removed
Checking for bi-allelic SNPs.
347,753 SNPs are non-biallelic. These will be removed.
N already exists within sumstats_dt.
1,617,610 SNPs (13.7%) have FRQ values > 0.5. Conventionally the FRQ column is intended to show the minor/effect allele frequency.
The FRQ column was mapped from one of the following from the inputted  summary statistics file:
FRQ, EAF, MAF, FRQ_U, F_U, FREQUENCY, FREQ, FREQ_TESTED_ALLELE, FRQ_TESTED_ALLELE, FREQ_EFFECT_ALLELE, FRQ_EFFECT_ALLELE, EFFECT_ALLELE_FREQUENCY, EFFECT_ALLELE_FREQ, EFFECT_ALLELE_FRQ, A1FREQ, A1FRQ, A2FREQ, A2FRQ, ALLELE_FREQUENCY, ALLELE_FREQ, ALLELE_FRQ, AF, MINOR_AF, EFFECT_AF, A2_AF, EFF_AF, ALT_AF, ALTERNATIVE_AF, INC_AF, A_2_AF, TESTED_AF
As frq_is_maf=TRUE, the FRQ column will not be renamed. If the FRQ values were intended to represent major allele frequency,
set frq_is_maf=FALSE to rename the column as MAJOR_ALLELE_FRQ and differentiate it from minor/effect allele frequency.
Sorting coordinates.
Writing in tabular format ==> /var/folders/hd/jm8lzp7s4dl_wlkykzhz66x80000gn/T//Rtmp916qbx/file388825789f34.tsv.gz
Summary statistics report:                                                                                                           
   - 11,774,743 rows (85.4% of original 13,791,467 rows)
   - 11,773,861 unique variants
   - 0 genome-wide significant variants (P<5e-8)
   - 22 chromosomes
Successfully finished preparing sumstats file, preview:
Reading header.
           SNP CHR     BP A1 A2         FRQ         NOTE BETA SE   P      N
1: rs547522712   1  15791  C  T 0.00000e+00  1:15791:C:T    1  1 0.9 500000
2: rs568226429   1  69487  G  A 5.53719e-06  1:69487:G:A    1  1 0.9 500000
3:   rs2531267   1  69569  T  C 1.88264e-04  1:69569:T:C    1  1 0.9 500000
4: rs533633326   1 139853  C  T 5.53719e-06 1:139853:C:T    1  1 0.9 500000
Returning path to saved data.
```

**Comment by albert-ying** [comment]
Ah, I used the BiocManager version. The error indeed disappeared in the GitHub version. Sorry! But maybe worth to add this to the readme
```
# For dev version
remotes::install_github("neurogenomics/MungeSumstats")
```

**Comment by albert-ying** [comment]
Just a little warning on the Bioconductor version.

I noticed there are some errors in my recent analysis, and I noticed that these errors only happen to the GWAS data that I munged after I update to the Bioconductor version of the MungeSumstat.

Specifically, the current Bioconductor version flips the sign of BETA without flipping the effect allele and ref allele. Could you confirm it? I'm not sure where I should re-munge all my GWAS using the current version or just flip the BETA back again.

Also, some of my GWAS are munged in previous versions (earlier than the Bioconductor one), they don't have such errors, but are they going through the same procedure as the current version? would you recommend to re-munge all of them using current version?

**Comment by Al-Murphy** [comment]
Hey Albert,

By the Bioconductor version, I'm guessing you mean the current release of bioconductor which is 3.14 (not devel version which is 3.15), MungeSumstats 1.2.0 is in the current release of bioconductor. Do you mean  MungeSumstats 1.2.0 versus a previous version?

I ask since there was a bug with allele flipping but this was just when MungeSumstats was first released on bioconductor (version 1.1.2 fixed this), so if your version is after this, I don't think there should be a difference in how it handles allele flipping.

Can you send on an example SNP and explain what is happening to it with the MungeSumstats Bioconductor release version and what you expect to happen? If you are concerned about differences in different versions have a look at our [news file](https://github.com/neurogenomics/MungeSumstats/blob/master/NEWS.md) but overall, I would not expect major differences from 1.1.2 forward only enhancements and extra functionality.

**Comment by albert-ying** [comment]
Hmm seem that my Bioconductor version is 3.13 and somehow installed that first released version of MungeSumstats, which cause the bug.

Thank you so much for confirming!
